internal/linker/link_node_modules.ts

```diff
@@ -260,30 +260,45 @@ export async function main(args: string[], runfiles: Runfiles) {
   const links = [];
 
   const linkModule =
-      async (name: string, modulePath: string) => {
-    let target = runfiles.resolve(modulePath);
-
-    // It sucks that we have to do a FS call here.
-    // TODO: could we know which packages are statically linked??
-    if (!target || !await exists(target)) {
-      // Try the bin dir
+      async (name: string, kind: 'bin'|'src'|'runfiles', modulePath: string) => {
+    let target: string = '<package linking failed>';
+    if (kind === 'bin') {
+      // FIXME(#1196)
       target = path.join(workspaceRelative, bin, toWorkspaceDir(modulePath));
+      // Spend an extra FS lookup to give better error in this case
       if (!await exists(target)) {
-        // Try the execroot
-        target = path.join(workspaceRelative, toWorkspaceDir(modulePath));
+        // TODO: there should be some docs explaining how users are
+        // expected to declare ahead of time where the package is loaded,
+        // how that relates to npm link scenarios,
+        // and where the configuration can go.
+        return Promise.reject(`ERROR: no output directory found for package ${modulePath}
+        Did you mean to declare this as a from-source package?
+        See https://github.com/bazelbuild/rules_nodejs/pull/1197
+        until this feature is properly documented.`);
       }
     }
+    if (kind === 'src') {
```

Comment:
nit: `else if`

Reply:
`switch/case/break`!

Reply:
is there a widely accepted threshold? 2 or less use if/else if/else; 3 or more use switch/case/break/default?